lib/oli/analytics/datashop.ex

```diff
@@ -144,10 +142,7 @@ defmodule Oli.Analytics.Datashop do
         date: part_attempt.date_submitted,
         email: email,
         sub: sub,
-        # datashop_session_id will be set to the latest part attempt datashop_session_id
-        # unless it is nil, then it will be set to a generated UUID. This will handle any
-        # part attempts that existed before this value was tracked in the part attempt record
-        datashop_session_id: value_or(part_attempt.datashop_session_id, UUID.uuid4()),
```

Comment:
We should not be generating a datashop id on the fly if one doesn't exist. This could lead to non-deterministic and misleading data